include/libopencm3/nrf5/common/clock_common.h

```diff
@@ -0,0 +1,224 @@
+/** @defgroup clock_lfclkstat LFCLK Status
+@ingroup clock_defines
+@{*/
+#define CLOCK_LFCLKSTAT_SRC_RC                     (0 << 0)
+#define CLOCK_LFCLKSTAT_SRC_XTAL                   (1 << 0)
+#define CLOCK_LFCLKSTAT_SRC_Xtal                   (1 << 0)
+#define CLOCK_LFCLKSTAT_SRC_SYNTH                  (0x2 << 0)
+#define CLOCK_LFCLKSTAT_SRC_Synth                  (0x2 << 0)
+#define CLOCK_LFCLKSTAT_STATE_NOTRUNNING           (0 << 16)
+#define CLOCK_LFCLKSTAT_STATE_NotRunning           (0 << 16)
+#define CLOCK_LFCLKSTAT_STATE_RUNNING              (1 << 16)
+#define CLOCK_LFCLKSTAT_STATE_Running              (1 << 16)
+/**@}*/
+
+/* --- CLOCK_LFCLKSRCCOPY values ------------------------------------------- */
+/** @defgroup clock_lfclksrccopy Copy of LFCLKSRC register, set when 
+LFCLKSTART task was triggered
+@ingroup clock_defines
+@{*/
+#define CLOCK_LFCLKSRCCOPY_SRC_RC                  (0)
+#define CLOCK_LFCLKSRCCOPY_SRC_XTAL                (1)
+#define CLOCK_LFCLKSRCCOPY_SRC_SYNTH               (2)
+/**@}*/
+
+/* --- CLOCK_LFCLKSRC values ----------------------------------------------- */
+/** @defgroup clock_lfclksrc Clock source for the LFCLK
+@ingroup clock_defines
+@{*/
+#define RC                                         (0 << 0)
+#define CLOCK_LFCLKSRC_SRC_RC                      (0 << 0)
+#define XTAL                                       (1 << 0)
+#define CLOCK_LFCLKSRC_SRC_XTAL                    (1 << 0)
+#define SYNTH                                      (0x2 << 0)
+#define CLOCK_LFCLKSRC_SRC_SYNTH                   (0x2 << 0)
+#define CLOCK_LFCLKSRC_BYPASS_ENABLED              (1 << 16)
+#define CLOCK_LFCLKSRC_BYPASS_Enabled              (1 << 16)
+#define CLOCK_LFCLKSRC_EXTERNAL_ENABLED            (1 << 17)
+#define CLOCK_LFCLKSRC_EXTERNAL_Enabled            (1 << 17)
+/**@}*/
+
+/* --- CLOCK_CTIV values --------------------------------------------------- */
+/** @defgroup clock_ctiv Retained register. Calibration timer interval
+@ingroup clock_defines
+@{*/
+#define CLOCK_CTIV_CTIV(n)                         (n*4)
+/**@}*/
+
+/* --- CLOCK_TRACECONFIG values -------------------------------------------- */
+/** @defgroup clock_traceconfig Retained register. Clocking options for the 
+Trace Port debug interface
+@ingroup clock_defines
+@{*/
+#define CLOCK_TRACECONFIG_TRACEPORTSPEED_32MHZ        (0 << 0) //TRACECLK = 16 MHz
+#define CLOCK_TRACECONFIG_TRACEPORTSPEED_16MHZ        (1 << 0) //TRACECLK = 8 MHz
+#define CLOCK_TRACECONFIG_TRACEPORTSPEED_8MHZ         (0x2 << 0) //TRACECLK = 4 MHz
+#define CLOCK_TRACECONFIG_TRACEPORTSPEED_4MHZ         (0x3 << 0) //TRACECLK = 2 MHz
+//gpio: GPIOs multiplexed onto all trace-pins
+#define CLOCK_TRACECONFIG_TRACEMUX_GPIO               (0 << 16) 
+//Serial: SWO multiplexed onto P0.18, GPIO multiplexed onto other trace-pins
+#define CLOCK_TRACECONFIG_TRACEMUX_SERIAL             (1 << 16) 
+//Parallel: TRACECLK and TRACEDATA multiplexed onto P0.20, P0.18, P0.16, P0.15, P0.14
```

Comment:
use /** here, so this useful documentation makes it into the documentation.
(same fo the traceclk / traceportspeed docs above)